src/session/sidebar/row.rs

```diff
@@ -219,45 +209,127 @@ impl Row {
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_status_stack();
+                obj.update_actions();
+                None
+            }),
+        );
+        chat_signal_group.connect_local(
+            "notify::unread-count",
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_status_stack();
+                obj.update_actions();
+                None
+            }),
+        );
+        chat_signal_group.connect_local(
+            "notify::unread-mention-count",
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_status_stack();
+                None
+            }),
+        );
+        chat_signal_group.connect_local(
+            "notify::last-read-outbox-message-id",
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_message_status_icon();
+                None
+            }),
+        );
+        chat_signal_group.connect_local(
+            "notify::last-message",
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_message_status_icon();
+                obj.update_timestamp();
+                obj.update_subtitle_prefix_label();
+                obj.update_minithumbnail();
+                obj.update_subtitle_label();
+                None
+            }),
+        );
+        chat_signal_group.connect_local(
+            "notify::draft-message",
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_timestamp();
+                obj.update_subtitle_prefix_label();
+                obj.update_minithumbnail();
+                obj.update_subtitle_label();
+                None
+            }),
+        );
+        chat_signal_group.connect_local(
+            "notify::actions",
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_subtitle_prefix_label();
+                obj.update_minithumbnail();
+                obj.update_subtitle_label();
+                None
             }),
-        )
-        .bind(&*imp.status_stack, "visible-child", Some(self));
+        );
+        chat_signal_group.connect_local(
+            "notify::notification-settings",
+            false,
+            clone!(@weak self as obj => @default-return None, move |_| {
+                obj.update_unread_count_style();
+                None
+            }),
+        );
+        imp.chat_signal_group.set(chat_signal_group).unwrap();
+
+        let session_signal_group = glib::SignalGroup::new(Session::static_type());
+        session_signal_group.connect_local(
+            "notify::private-chats-notification-settings",
+            false,
```

Comment:
```suggestion
        session_signal_group.connect_notify_local(
            Some("private-chats-notification-settings"),
```

and elsewhere maybe? :)

Reply:
Ah! No idea how I didn't notice that method. Thanks!

Reply:
Oh wait, that doesn't work. This is confusing, but `connect_notify_local` is a blanket implementation of `ObjectExt` of the signal group, so it's actually useless because it listens to the notify signal of the `SignalGroup`'s properties instead of the target one. See https://gtk-rs.org/gtk-rs-core/stable/latest/docs/glib/struct.SignalGroup.html#method.connect_local.

Maybe that's something that could be added to `gtk-rs-core`?

Reply:
Reverted for now.